src/awkward/_v2/_util.py

```diff
@@ -1123,7 +1123,7 @@ def _impl(array):
         elif isinstance(array, ak._v2.contents.IndexedOptionArray):
             content = _impl(array.project())
 
-            mask0 = module.asarray(array.bytemask()).view(np.bool_)
+            mask0 = array.mask_as_bool(valid_when=False)
```

Comment:
This last case illustrates why we want to avoid having both `bytemask` and `mask_as_bool`. You didn't see it, @ioanaif, because it was added by @swishdiff, and @swishdiff didn't know that `bytemask` was being removed.